server/schedulers/hot_region.go

```diff
@@ -903,3 +1041,27 @@ func (ty opType) String() string {
 		return ""
 	}
 }
+
+type resourceType int
+
+const (
+	writePeer resourceType = iota
+	writeLeader
+	readLeader
+	resourceTypeLen
+)
+
+func toResourceType(rwTy rwType, opTy opType) resourceType {
+	switch rwTy {
+	case write:
+		switch opTy {
+		case movePeer:
+			return writePeer
+		case transferLeader:
+			return writeLeader
+		}
+	case read:
+		return readLeader
+	}
+	return resourceTypeLen
```

Comment:
Will it cause panic?

Reply:
It will only cause panic when the arguments are invalid. I think this is acceptable.

Reply:
It's better to panic in this function rather than use it?

Reply:
Fixed.